File: app/src/main/java/com/degree/application/view/MainActivity.java
package com.degree.application.view;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v4.widget.NestedScrollView;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.MotionEvent;
import android.view.View;
import android.widget.EditText;

import com.degree.application.R;
import com.degree.application.view.fragment.CheckFragment;
import com.degree.application.view.fragment.ScanFragment;
import com.degree.application.view.fragment.StatusFragment;
import com.shuhart.stepview.StepView;

import static com.degree.application.service.ContractContext.setIp;

public class MainActivity extends AppCompatActivity implements ViewPager.OnPageChangeListener, View.OnTouchListener {

    private StepView stepView;
    private SectionsPagerAdapter mSectionsPagerAdapter;
    private ViewPager mViewPager;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        NestedScrollView scrollView = findViewById(R.id.nested_scroll_view);
        scrollView.setFillViewport(true);

        stepView = findViewById(R.id.step_view);

        // Create the adapter that will return a fragment for each of the three
        // primary sections of the activity.
        mSectionsPagerAdapter = new SectionsPagerAdapter(getSupportFragmentManager());

        // Set up the ViewPager with the sections adapter.
        mViewPager = findViewById(R.id.container);
        mViewPager.setAdapter(mSectionsPagerAdapter);
        mViewPager.addOnPageChangeListener(this);
        mViewPager.setOnTouchListener(this);
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_ip:
                showDialogConfiguration(item.getItemId());
                // About option clicked.
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    private void showDialogConfiguration(int id) {
        EditText taskEditText = new EditText(this);
        AlertDialog dialog = new AlertDialog.Builder(this)
                .setTitle(R.string.dialog_set_configuration)
                .setMessage(R.string.dialog_set_configuration_message)
                .setView(taskEditText)
                .setPositiveButton(R.string.dialog_set_configuration_ok, (dialog1, which) -> {
                    setIp(taskEditText.getText().toString());
                })
                .setNegativeButton(R.string.dialog_set_configuration_cancel, null)
                .create();
        dialog.show();
    }


    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

    }

    @Override
    public void onPageSelected(int position) {
        stepView.go(position, true);
    }

    @Override
    public void onPageScrollStateChanged(int state) {

    }

    @Override
    public boolean onTouch(View v, MotionEvent event) {
        return true;
    }

    public void firstItem() {
        mViewPager.setCurrentItem(0);
    }

    public void nextCurrentItem() {
        mViewPager.setCurrentItem(mViewPager.getCurrentItem() + 1);
    }

    /**
     * A {@link FragmentPagerAdapter} that returns a fragment corresponding to
     * one of the sections/tabs/pages.
     */
    public class SectionsPagerAdapter extends FragmentPagerAdapter {

        public SectionsPagerAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {
            // getItem is called to instantiate the fragment for the given page.
            // Return a PlaceholderFragment (defined as a static inner class below).
            switch (position) {
                case 0:
                default:
                    return ScanFragment.newInstance();
                case 1:
                    return CheckFragment.newInstance();
                case 2:
                    return StatusFragment.newInstance();
            }
        }

        @Override
        public int getCount() {
            // Show 3 total pages.
            return 3;
        }
    }
}

File: app/src/main/java/io/smartpal/smartdegree/view/activity/MainActivity.kt
package io.smartpal.smartdegree.view.activity

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.ViewPager
import android.support.v4.widget.NestedScrollView
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.view.MotionEvent
import android.view.View
import io.smartpal.smartdegree.R
import io.smartpal.smartdegree.view.fragment.CheckFragment
import io.smartpal.smartdegree.view.fragment.ScanFragment
import io.smartpal.smartdegree.view.fragment.StatusFragment
import com.shuhart.stepview.StepView

class MainActivity : AppCompatActivity(), View.OnTouchListener, ViewPager.OnPageChangeListener {

    private lateinit var stepView: StepView
    private lateinit var mSectionsPagerAdapter: SectionsPagerAdapter
    private lateinit var mViewPager: ViewPager


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setupToolbar()
        setupContentView()
        setupStepView()


    }

    private fun setupContentView() {
        findViewById<NestedScrollView>(R.id.nested_scroll_view).isFillViewport = true
    }

    private fun setupToolbar() {
        setSupportActionBar(findViewById(R.id.toolbar))
    }

    private fun setupStepView() {
        stepView = findViewById(R.id.step_view)
        mViewPager = findViewById(R.id.container)

        // Create the adapter that will return a fragment for each of the three
        // primary sections of the activity.
        mSectionsPagerAdapter = SectionsPagerAdapter(supportFragmentManager)

        // Set up the ViewPager with the sections adapter.
        mViewPager.adapter = mSectionsPagerAdapter
        mViewPager.addOnPageChangeListener(this)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_navigation, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_settings -> {
                startActivity(Intent(this,SettingsActivity::class.java))
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    /**
     * A [FragmentPagerAdapter] that returns a fragment corresponding to
     * one of the tabs.
     */
    inner class SectionsPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

        override fun getItem(position: Int): Fragment {
            // getItem is called to instantiate the fragment for the given page.
            // Return a PlaceholderFragment (defined as a static inner class below).


            return when (position) {
                0 ->  ScanFragment.newInstance()
                1 -> CheckFragment.newInstance()
                2 -> StatusFragment.newInstance()
                else -> CheckFragment.newInstance()
            }
        }

        override fun getCount(): Int {
            // Show 3 total pages.
            return 3
        }
    }

    fun firstItem() {
        mViewPager.currentItem = 0
    }

    fun nextItem() {
        mViewPager.currentItem = mViewPager.currentItem + 1
    }


    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
        stepView.go(position, true)
    }

    override fun onPageScrollStateChanged(state: Int) = Unit

    override fun onPageSelected(position: Int) = Unit

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouch(view: View?, p1: MotionEvent?): Boolean {
       return true
    }


}